darwin/cli_functions.py

```diff
@@ -927,6 +937,66 @@ def delete_files(dataset_slug: str, files: List[str], skip_user_confirmation: bo
         _error(f"An error has occurred, please try again later.")
 
 
+def validate_schemas(
+    files: Optional[List[str]] = None,
+    folder: Optional[str] = None,
+    pattern: Optional[str] = None,
+    silent: bool = False,
+    output: Optional[Path] = None,
+):
+    """
+    Validate function for the CLI. Takes one of 3 required key word arguments describing the location of files and prints and/or saves an output
+
+    Parameters
+    ----------
+    files : Optional[List[str]], optional
+        list of file paths to check, by default None
+    folder : Optional[str], optional
+        folder location to check for *.json files, by default None
+    pattern : Optional[str], optional
+        glob style pattern matching, by default None
+    silent : bool, optional
+        flag to set silent console printing, only showing errors, by default False
+    output : Optional[Path], optional
+        filename for saving to output, by default None
+    """
+    if not files and not folder and not pattern:
+        # TODO insert logging here once we introduce that
```

Comment:
In the meantime, this is a CLI command, so move L974 above this and
```suggestion
        console.log('--files or --folder flag requires a pattern')
```
to give the user firm output on their issue?

https://rich.readthedocs.io/en/stable/console.html#attributes - Rich has a lot of stuff for this, and it can be handy.

Reply:
I'm also pretty sure that the `argparse` parsing actually makes this condition impossible, but handling it is good to avoid someone breaking it in future.